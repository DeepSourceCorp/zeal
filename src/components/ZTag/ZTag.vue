<template>
  <div class="bg-ink-300 inline-flex items-center justify-evenly text-vanilla-100 rounded-full space-x-2"
        :class="[ getSizeBasedClasses,
                  getBaseColor]">
    <z-icon v-if="iconLeft" :icon="iconLeft" :size="size"></z-icon>
    <div><slot></slot></div>
    <z-icon v-if="iconRight" :icon="iconRight" :size="size"></z-icon>
  </div>
</template>

<script lang="ts">
import ZIcon from '@/components/ZIcon'
export default {
  name: 'ZTag',
  components: {
    ZIcon
  },
  props: {
    state: {
        type: String,
        default: 'default',
        validator: function (value: string): boolean {
            return ['success', 'error', 'warning', 'info', 'default'].includes(value)
        }
    },
    iconLeft: {
        type: String
    },
    iconRight: {
        type: String
    },
    size: {
        type: String,
        default:'small',
        validator: function (value: string): boolean {
            return ['small', 'large'].includes(value)
        }
    }
  },
  computed: {
    getSizeBasedClasses(): string {
        const sizes = {
            small: 'py-1 px-4 text-sm',
            large: 'py-1 px-4 text-lg'
        }
        return sizes[this.size];
    },
    getBaseColor(): string {
        const states = {
            'success': 'border-2 border-solid border-juniper',
            'default': '',
            'info': 'border-2 border-solid border-aqua',
            'warning': 'border-2 border-solid border-honey',
            'error': 'border-2 border-solid border-cherry',
        }
        return states[this.state]
    }
  }
}
</script>
