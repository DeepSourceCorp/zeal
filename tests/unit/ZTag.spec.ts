import ZTag from '../../src/components/ZTag'

import { shallowMount } from '@vue/test-utils'

const slotData = {
  default: 'Tag'
}

describe('Tags', () => {
  it('renders a default tag', () => {
    const wrapper = shallowMount(ZTag, {
      slots: slotData
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders with left icon', () => {
    const wrapper = shallowMount(ZTag, {
      propsData: {
        iconLeft: 'star'
      },
      slots: slotData
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders with right icon', () => {
    const wrapper = shallowMount(ZTag, {
      propsData: {
        iconRight: 'star'
      },
      slots: slotData
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders with state', () => {
    expect(
      shallowMount(ZTag, {
        propsData: {
          state: 'success'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()

    expect(
      shallowMount(ZTag, {
        propsData: {
          state: 'error'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()

    expect(
      shallowMount(ZTag, {
        propsData: {
          state: 'warning'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()

    expect(
      shallowMount(ZTag, {
        propsData: {
          state: 'info'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()

    expect(
      shallowMount(ZTag, {
        propsData: {
          state: 'default'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()
  })

  it('renders with size', () => {
    expect(
      shallowMount(ZTag, {
        propsData: {
          iconLeft: 'star',
          size: 'small'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()

    expect(
      shallowMount(ZTag, {
        propsData: {
          iconLeft: 'star',
          size: 'large'
        },
        slots: slotData
      }).html()
    ).toMatchSnapshot()
  })
})
