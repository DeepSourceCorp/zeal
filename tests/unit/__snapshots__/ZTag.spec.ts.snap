// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders a default tag 1`] = `
<div class="inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 py-1 px-4 text-sm bg-ink-300">
  <!----> Tag
  <!---->
</div>
`;

exports[`Tags renders with left icon 1`] = `
<div class="inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 py-1 px-4 text-sm bg-ink-300">
  <z-icon-stub icon="star" size="small" position="" color="current"></z-icon-stub> Tag
  <!---->
</div>
`;

exports[`Tags renders with right icon 1`] = `
<div class="inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 py-1 px-4 text-sm bg-ink-300">
  <!----> Tag <z-icon-stub icon="star" size="small" position="" color="current"></z-icon-stub>
</div>
`;

exports[`Tags renders with size 1`] = `
<div class="inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 py-1 px-4 text-sm bg-ink-300">
  <z-icon-stub icon="star" size="small" position="" color="current"></z-icon-stub> Tag
  <!---->
</div>
`;

exports[`Tags renders with size 2`] = `
<div class="inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 py-1 px-4 text-sm bg-ink-300">
  <z-icon-stub icon="star" size="large" position="" color="current"></z-icon-stub> Tag
  <!---->
</div>
`;

exports[`Tags renders with state 1`] = `
<div class="inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 py-1 px-4 text-sm border-2 border-solid border-juniper bg-ink-300">
  <!----> Tag
  <!---->
</div>
`;

exports[`Tags renders with state 2`] = `
<div class="inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 py-1 px-4 text-sm border-2 border-solid border-cherry bg-ink-300">
  <!----> Tag
  <!---->
</div>
`;

exports[`Tags renders with state 3`] = `
<div class="inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 py-1 px-4 text-sm border-2 border-solid border-honey bg-ink-300">
  <!----> Tag
  <!---->
</div>
`;

exports[`Tags renders with state 4`] = `
<div class="inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 py-1 px-4 text-sm border-2 border-solid border-aqua bg-ink-300">
  <!----> Tag
  <!---->
</div>
`;

exports[`Tags renders with state 5`] = `
<div class="inline-flex items-center space-x-2 rounded-full justify-evenly text-vanilla-100 py-1 px-4 text-sm bg-ink-300">
  <!----> Tag
  <!---->
</div>
`;
